dimod/reference/composites/fixedvariable.py

```diff
@@ -0,0 +1,103 @@
+import dimod
+from dimod.roof_duality import fix_variables
+from dimod.response import SampleSet
+import numpy as np
+
+__all__ = ['FixedVariableComposite']
+
+class FixedVariableComposite(dimod.ComposedSampler):
+
+    def __init__(self, child_sampler):
+        self._children = [child_sampler]
+
+    @property
+    def children(self):
+
+        return self._children
+
+    @property
+    def parameters(self):
+
+        return self.child.parameters.copy()
+
+    @property
+    def properties(self):
+
+        return {'child_properties': self.child.properties.copy()}
+
+    def sample(self, source_bqm, fixed_variables=None,
+               **parameters):
+
+        child = self.child
+        if fixed_variables is None:
+            fixed_variables = fix_variables(source_bqm)
+        bqm = source_bqm.copy()
+
+        for v, val in fixed_variables.items():
+            bqm.fix_variable(v, val)
+
+        if len(bqm.variables)==0:
+            return _fixed_response(fixed_variables,bqm.offset,bqm.vartype)
```

Comment:
couldn't you just generate an empty response? Then pass it through `_release_response`?

Reply:
i feel like this is more efficient... but may not be.

Reply:
It is, but for a case that's silly. So I don't feel like we have a responsibility to optimize it....

Reply:
its written already?